src/poll.rs

```diff
@@ -179,6 +179,180 @@ libc_bitflags! {
     }
 }
 
+/// Timeout argument for [`poll`].
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct PollTimeout(i32);
+
+impl PollTimeout {
+    /// Blocks indefinitely.
+    ///
+    /// > Specifying a negative value in timeout means an infinite timeout.
+    pub const NONE: Self = Self(-1);
+    /// Returns immediately.
+    ///
+    /// > Specifying  a timeout of zero causes poll() to return immediately, even if no file
+    /// > descriptors are ready.
+    pub const ZERO: Self = Self(0);
+    /// Blocks for at most [`std::i32::MAX`] milliseconds.
+    pub const MAX: Self = Self(i32::MAX);
+    /// Returns if `self` equals [`PollTimeout::NONE`].
+    pub fn is_none(&self) -> bool {
+        // > Specifying a negative value in timeout means an infinite timeout.
+        *self <= Self::NONE
+    }
+    /// Returns if `self` does not equal [`PollTimeout::NONE`].
+    pub fn is_some(&self) -> bool {
+        !self.is_none()
+    }
+    /// Returns the timeout in milliseconds if there is some, otherwise returns `None`.
+    pub fn timeout(&self) -> Option<i32> {
```

Comment:
I think naming this method `as_millis` would be more idiomatic.  For example, it would mirror `Duration::as_millis`.  And since it will never return a negative value, should it return `u32` instead of `i32`?

Reply:
I've split this into 2 functions
```rust
    /// Returns the timeout in milliseconds if there is some, otherwise returns `None`.
    pub fn as_millis(&self) -> Option<i32> {
        self.is_some().then_some(self.0)
    }
    /// Returns the timeout as a `Duration` if there is some, otherwise returns `None`.
    pub fn timeout(&self) -> Option<Duration> {
        self.as_millis().map(|x|Duration::from_millis(u64::try_from(x).unwrap()))
    }
```

Reply:
I think using `i32` or `u32` each have their own drawbacks here, ideally we could use `u31` with something like https://crates.io/crates/ux but this crate is not actively maintained. I tried my hand at making a better version as an experiment with https://crates.io/crates/ux2 but I am busy so am not able to maintain this alone.

Reply:
I agree that `u31` suits best here, but before switching to the `ux` or `ux2` crate, `u32` is the better choice compared to `i32`

Reply:
What about this one?

Reply:
I am okay to use `u32`.

Reply:
Updated to u32.